pkg/systemd/services/timer-dialog.jsx

```diff
@@ -268,49 +398,72 @@ const CreateTimerDialogBody = ({ owner }) => {
-                            const validationFailedSecond = !(/^[0-9]+$/.test(sec) && sec <= 59 && sec >= 0);
-
-                            if (submitted && repeat == 'minutely' && validationFailedSecond) {
-                                validated = "error";
-                                helperTextInvalid = _("Second needs to be a number between 0-59");
-                            }
+                            const repeatValidate = (event) => {
+                                if (!event) return;
+                                let helperTextInvalid;
+                                switch (event.target.id) {
+                                case 'hourly': {
+                                    helperTextInvalid = _("Minute needs to be a number between 0-59");
+                                    break;
+                                }
+                                case 'minutely': {
+                                    helperTextInvalid = _("Second needs to be a number between 0-59");
+                                    break;
+                                }
+                                case 'yearly': {
+                                    helperTextInvalid = _("Please enter valid date.  Example YYYY-MM-DD");
+                                    break;
+                                }
+                                }
+                                const validated = !event.target.checkValidity() ? "error" : "default";
+                                const arr = [...repeatPatterns];
+                                arr[idx].helperTextInvalid = helperTextInvalid;
+                                arr[idx].validated = validated;
+                                setRepeatPatterns(arr);
+                                if (validated == 'default') {
+                                    setDialogError();
+                                    formCheck();
+                                }
+                            };
 
                             return (
                                 <FormGroup label={label} key={item.key}
-                                           validated={validated}
-                                           helperTextInvalid={helperTextInvalid}>
+                                           validated={repeatPatterns[idx].validated}
+                                           helperTextInvalid={repeatPatterns[idx].helperTextInvalid}>
                                     <Flex className="specific-repeat-group" data-index={idx}>
                                         {repeat == "minutely" &&
                                             <TextInput className='delay-number'
                                                        id={repeat}
+                                                       isRequired
                                                        value={repeatPatterns[idx].second}
+                                                       onBlur={repeatValidate}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)